src/compiler/checker.ts

```diff
@@ -24849,18 +24907,136 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
         }
 
+        /**
+         * Semantically starts a new group of alternatives within the active inference context
+         */
+        function forkInferenceContext(): InferenceContext[] {
+            return [];
+        }
+
+        /**
+         * Performs the given action in every alternative inference context currently under consideration,
+         * and add the resulting context(s) to a new resultant context list.
+         *
+         * Call multiple times to create multiple independent forks of the existing inference engine state.
+         * Call once with fork `false` to simply to do and action for every existing branch of the inference

```

Comment:
"to simply to do and action for every [...]" - I'm not sure how to read this properly, perhaps this part could be improved/rephrased?